web/app/components/dropdown/dropdown.jsx

```diff
@@ -0,0 +1,81 @@
+import { h, Component } from 'preact';
+
+import Button from '../button';
+
+export default class Dropdown extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      isActive: props.isActive || false,
+    };
+
+    this.onClickTitle = this.onClickTitle.bind(this);
+    this.onOutsideClick = this.onOutsideClick.bind(this);
+  }
+
+  bindRef(field) {
+    return (ref) => this[field] = ref;
+  }
+
+  onClickTitle() {
+    this.setState({
+      isActive: !this.state.isActive
+    });
+
+    if (this.props.onClickTitle) {
+      this.props.onClickTitle();
+    }
+  }
+
+  onOutsideClick(e) {
+    if (!this._dropdown.contains(e.target) && e.target !== this._dropdown) {
+      if (this.state.isActive) {
+        this.setState({
+          isActive: false,
+        });
+      }
+    }
+  }
+
+  componentDidMount() {
+    document.addEventListener('click', this.onOutsideClick);
+    parent && parent.document.addEventListener('click', this.onOutsideClick);
+  }
+
+  componentWillUnmount() {
+    document.removeEventListener('click', this.onOutsideClick);
+    parent && parent.document.removeEventListener('click', this.onOutsideClick);
+  }
+
+  render(props, { isActive }) {
+    const { title, heading, children } = props;
+
+    return (
+      <div className={b('dropdown', { mods: { isActive }})} ref={this.bindRef('_dropdown')}>
```

Comment:
Let's use clearer code for refs? Like `ref={r => (this.rootNode = r)}`. 

I mean, right now you can't understand this code without reading the `bindRef` definition. But it's the strange wrapper and you can omit it, can't you?